server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -468,19 +468,18 @@ export class AgenticChatController implements ChatHandlers {
      * @param session
      */
     async waitForToolApproval(
-        toolUseId: string,
-        toolUseName: string,
+        toolUse: ToolUse,
         resultStream: AgenticChatResultStream,
         promptBlockId: number,
         session: ChatSessionService
     ) {
         const deferred = this.#createDeferred()
-        session.setDeferredToolExecution(toolUseId, deferred.resolve, deferred.reject)
-        this.#log(`Prompting for tool approval for tool: ${toolUseName}`)
+        session.setDeferredToolExecution(toolUse.toolUseId!, deferred.resolve, deferred.reject)
```

Comment:
is there anyway we can avoid the type assertion here? Perhaps by requiring it in the params?

Reply:
will followup